### PR TITLE
Read api key from ~/.wakatime.cfg file

### PR DESCRIPTION
**Pull request by jvelezpo**
* the code to read from `.wakatime.cfg` was taken from `vscode-wakatime`
* Fixes some linter issues
* Dependencies now dont use static versioning, so it updates easier

### CODE REVIEW

File: src/utils/options.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  constructor() {
+    this.cache = new ExpirationStrategy(new MemoryStorage());
+    const wakaHome = this.getWakaHome();
+    this.configFile = path.join(wakaHome, '.wakatime.cfg');
+    this.logFile = path.join(wakaHome, '.wakatime.log');
+  }
+
+  private getWakaHome(): string {
+    const home = process.env.WAKATIME_HOME;
```
Comment (gandarez) [inline src/utils/options.ts:23]: Sometimes WAKATIME_HOME variable is set using the symbol `~` and/or with trailing slash at the end like `~/mypath` or `~/mypath/`. It needs to be expanded before returning. Does NodeJS resolve "~"?

File: src/utils/options.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  public getConfigFile(): string {
+    return this.configFile;
+  }
+
+  public getLogFile(): string {
+    return this.logFile;
+  }
+
+  public async getApiKeyAsync(): Promise<string> {
+    const cachedApiKey = await this.cache.getItem<string>('api_key');
```
Comment (gandarez) [inline src/utils/options.ts:80]: There're legacy versions that still use `key` or `apikey` instead of `api_key`.

File: src/utils/options.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    if (cachedApiKey) {
+      return cachedApiKey;
+    }
+
+    const apiKey = await this.getSettingAsync<string>('settings', 'api_key');
```
Comment (gandarez) [inline src/utils/options.ts:85]: See my comment above.

File: src/utils/options.ts (hunk 1 above)
Comment (jvelezpo) [reply src/utils/options.ts:23]: if you go to the `constructor` when we use the `wakaHome` on path.join node normalizes the path so the situation is been handle there 😎

File: src/utils/options.ts [hunk 4]
```diff
@@ -81,9 +81,23 @@ export class Options {
...
+    try {
+      apiKey = await this.getSettingAsync<string>('settings', 'api_key');
+    } catch (_) {
+      try {
+        apiKey = await this.getSettingAsync<string>('settings', 'apikey');
+      } catch (__) {
+        try {
+          apiKey = await this.getSettingAsync<string>('settings', 'key');
+        } catch (err) {
+          console.log('api_key is not set in the system');
```
Comment (gandarez) [inline src/utils/options.ts:94]: maybe the message might be -> `api key not found on config file`

**Pull request merged**